src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -966,6 +966,7 @@ export class CodeWindow extends BaseWindow implements ICodeWindow {
 				newHttpProxy = newHttpProxy.substr(0, newHttpProxy.length - 1);
 			}
 
+			process.env['no_proxy'] = process.env['no_proxy'] || (this.configurationService.getValue<string[]>('http.noProxy') || ['']).map((item) => item.trim()).join(',')
```

Comment:
This is slightly inconsistent with how we handle the `http.proxy` setting. I suggest to leave the env variable as-is and pass the `http.noProxy` setting in proxyResolver.ts the same way the other settings are already passed.

Reply:
Thanks @chrmarti for the review. I have created a new PR for vscode-proxy-agent https://github.com/microsoft/vscode-proxy-agent/pull/35 to pass the setting in proxyResolver.ts to proxy agent